assets/adr/001-spec-transfer.md

```diff
@@ -0,0 +1,78 @@
+
+The `backends`, `diagnostics`, `loggers`, and `named values` folders in `/apimartifacts` folder do not need to be updated on new spec changes. The backend url for the AKS cluster does not change on spec updates and the monitoring configuration for the previous versions/revisions of the API will be applied to the new version/revision as well.
+
+In total, the files that should be added to the `/apimartifacts` folder when the backend API team releases a new spec are:
+
+- `/apis/{API_NAME}/apiInformation.json`
+- `/apis/{API_NAME}/policy.xml`
+- `/apis/{API_NAME}/specification.yaml`
+
+And the files that should potentially be updated are:
+
+- `/products/{PRODUCT_NAME}/apis.json`
+
+As mentioned, `specification.yaml` comes directly from the backend API team, as it is the new Open API Spec that must be applied to API Management. `/apis/{API_NAME}/apiInformation.json` and `/apis/{API_NAME}/policy.xml` should be automatically generated, and `/products/{PRODUCT_NAME}/apis.json` should be automatically updated.
+
+To generate the new files, we propose that we templatize the three files and create a parameters file that accompanies each release. The backend API team should release a parameters file that accompanies the Open API Spec, that when applied to the templates, generates the right API configuration for the new API version/revision. These files are extremely similar for all APIs, and the properties that differ between them be set via parameter. In the case of `apiInformation.json`, for instance, the `apiVersion`,`apiRevision`, and `apiRevisionDescription` properties should be templatized and set via parameters. The parameters file should additionally designate which API is being updated and whether it is a new version or revision.
+
+The incoming Open API spec and parameters file should trigger automatic generation of the `/apis/{API_NAME}/specification.yaml`, `/apis/{API_NAME}/apiInformation.json`, `/apis/{API_NAME}/policy.xml`, and the `/products/{PRODUCT_NAME}/apis.json` files in `/apimartifacts`, then publish the new configuration to the series of API Management instances, starting with `dev`. We propose we use [envsubst](https://linux.die.net/man/1/envsubst) to handle creation of those files from the templates and parameters files. The outputted files will be merged into the `/apimartifacts` in preparation for publish.
+
+The publish operations should be executed via container in the existing AKS cluster. In order to do so, we propose that we containerize the API Ops Publisher so it can be run inside AKS. An intermediate Azure Container Registry (ACR) should be used to store the container image for the Publisher to facilitate install in AKS. The newly updated `/apimartifacts/` contents will be copied into the Publisher's AKS pod in preparation for publish, along with the relevant environment variables necessary for publish to each specific environment (detailed in [README](../../README.md#create-github-environments)).
```

Comment:
```suggestion
The publish operations should be executed via container in the existing AKS cluster. In order to do so, we propose that we containerize the API Ops Publisher so it can be run inside AKS. An intermediate Azure Container Registry (ACR) should be used to store the container image for the Publisher to facilitate install in AKS. The newly updated `/apim_artifacts/` contents will be copied into the Publisher's AKS pod in preparation for publish, along with the relevant environment configuration file to publish to each specific environment (detailed in [README](../../README.md#create-github-environments)).
```